algorithms/ar-qsrt/java/QuickSort.java

```diff
@@ -0,0 +1,55 @@
+public class QuickSort {
+
+    public static void sort(int[] arr) {
+        if (arr == null || arr.length == 0) return;
+
+        quicksort(0, arr.length - 1, arr);
+    }
+
+    private static void quicksort(int left, int right, int[] arr) {
+        if (left >= right) return;
+
+        int mid = (left + right) / 2;
+        int pivot = arr[mid];
+
+        int partitionIndex = partition(left, right, pivot, arr);
+        quicksort(left, partitionIndex - 1, arr);
+        quicksort(partitionIndex, right, arr);
+    }
+
+    private static int partition(int left, int right, int pivot, int[] arr) {
```

Comment:
Codacy found an issue: [Avoid reassigning parameters such as 'left'](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6321063)

Reply:
@RavinduSachintha Should I create temporary variables to resolve the Codacy issue or is it okay?

Reply:
@mertbarutcuoglu Use temporary local variables.

Reply:
@Danushka96 Done.